eras/conway/impl/cddl-files/conway.cddl

```diff
@@ -51,38 +88,43 @@ big_nint = #6.3(bounded_bytes)
 
 big_uint = #6.2(bounded_bytes)
 
-block = [header
-         , transaction_bodies : [* transaction_body]
-         , transaction_witness_sets : [* transaction_witness_set]
-         , auxiliary_data_set : {* transaction_index => auxiliary_data}
-         , invalid_transactions : [* transaction_index]]
+; Valid blocks must also satisfy the following two constraints:
+;     1) the length of transaction_bodies and transaction_witness_sets
+;        must be the same
+;     2) every transaction_index must be strictly smaller than the
+;        length of transaction_bodies
+block = [header,
+         transaction_bodies : [* transaction_body],
+         transaction_witness_sets : [* transaction_witness_set],
+         auxiliary_data_set : {* transaction_index => auxiliary_data},
+         invalid_transactions : [* transaction_index]]
 
 block_no = uint .size 8
 
-bootstrap_witness = [public_key : $vkey
-                     , signature : $signature
-                     , chain_code : bytes .size 32
-                     , attributes : bytes]
+bootstrap_witness = [public_key : $vkey,
```

Comment:
@nc6, please revert the change that switched the pretty printing in cuddle from leading to trailing.

I know you personal preference, but it does not align with the previous formatting and with our preference.

```suggestion
bootstrap_witness = [ public_key : $vkey
                    , signature : $signature
```



Other things like
```
certificate = [stake_registration
                // stake_deregistration
``
are better than
```
certificate = [stake_registration // 
               stake_deregistration // 
```

Reply:
I have no personal preference here, other than for consistency. The switch to trailing commas was an attempt to better handle this scenario:

```
voter = [0, addr_keyhash
          // 1, scripthash
          // 2, addr_keyhash
          // 3, scripthash
          // 4, addr_keyhash]

voter = [0, addr_keyhash // 
         1, scripthash // 
         2, addr_keyhash // 
         3, scripthash // 
         4, addr_keyhash]
```

With the trailing commas and `//`, I could get the indices all lined up nicely. Honestly I didn't make a conscious decision to change things, I just tried a few formatting options until one looked nice enough. If you really have a strong preference for leading commas in CDDL, I can try to find something nice with that.

Reply:
With respect to commas, I really do have strong opinion for leading commas.

With respect to `//` we can keep it trailing, if that is easier, but if we can have this formatting instead, that would be even better:

```
voter = [  0, addr_keyhash
        // 1, scripthash
        // 2, addr_keyhash
        // 3, scripthash
        // 4, addr_keyhash
        ]
```

This would work much nicer with comments when get them implemented (this is a pretty high priority, since users are complaining about lack thereof), eg:

```
voter = [  0, addr_keyhash ; Committee Member Hot KeyHash
        // 1, scripthash   ; Committee Member Hot ScriptHash
        // 2, addr_keyhash ; DRep KeyHash
        // 3, scripthash   ; DRep ScriptHash
        // 4, addr_keyhash ; StakePool KeyHash
        ]
```

Reply:
Now updated with leading commas